HIP/hip-657.md

```diff
@@ -0,0 +1,115 @@
+---
+hip: 657
+title: Mutable metadata fields for dynamic NFTs
+author: Patches Punks <@HGraphPunks>
+working-group: Jason Fabritz <@bugbytesinc>, Sean Cuscus <@seancuscus>, Cooper Kunz <@Cooper-Kunz>, May Chan <@rocketmay>, Burstall Stowerling <@Burstall>, Ian Holsman <@web3-nomad>, Ashe Oro <@Ashe-Oro>, Michael Garber <@mgarbs>, Paul Sullivan <@paulgs9988>
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Review
+created: 2023-01-04
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/607
+updated: 2023-01-21, 2023-01-24
+---
+
+
+## Abstract
+
+
+Currently, there is no native way to update the metadata of a non-fungible token (NFT) on the Hedera network. This HIP proposes the creation of a new key called the METADATA key, which will allow updating NFT metadata. This will provide the Hedera network with functional parity with other networks and enable developers to create dynamic NFTs with new interactions.
+
+
+## Motivation
+
+
+Dynamic NFTs are a valuable addition to the world of non-fungible tokens because they allow for more interactive and immersive experiences. With the ability to update the metadata of an NFT, developers can create game pieces that gain new equipment and stats, legal agreements that reflect changing parameters, or even NFTs that update their images based on the time of day. Dynamic NFTs also open up the possibility for NFTs to be used in a wider range of industries and applications, such as supply chain tracking, real estate, among others. The potential for dynamic NFTs is numerous, and their introduction will greatly enhance the capabilities of the NFT ecosystem.
+
+
+## Rationale
+
+
+To address this issue, the proposed solution is to create a new key type called the METADATA key. This key will allow NFTs in a collection to update their metadata.
```

Comment:
Is the implication a new key type [here](https://github.com/hashgraph/hedera-protobufs/blob/main/services/basic_types.proto#L576)?

Reply:
no, not a new key type but rather the addition of a new purposeful key on the token_id called `METADATA`. It can be of the same type(s) as the other keys on the token_id as currently defined.

Reply:
Right, category of type in this context being: SUPPLY, WIPE, ADMIN, etc. Not the way the key is generated or its encryption / encoding.

Reply:
I can remove the word type because I can see how it might get misconstrued